Description: Cleanup imapproxy.conf
Author: Richard Laager <rlaager@wiktel.com>
Forwarded: https://sourceforge.net/p/squirrelmail/mailman/message/35543883/
Last-Update: 2016-11-30
--- a/scripts/imapproxy.conf
+++ b/scripts/imapproxy.conf
@@ -57,7 +57,7 @@
 ## server_port
 ##
 ## This setting specifies the port that server_hostname is listening on.
-## This is the tcp port that we proxy inbound connections to.
+## This is the TCP port that we proxy inbound connections to.
 ##
 ## If you are using SSL with IMAP Proxy, note that unless the server is
 ## highly non-standard, this should still be set to the server's normal,
@@ -118,7 +118,7 @@
 ##
 ## The logging facility to be used for all syslog calls.  If nothing is
 ## specified here, it will default to LOG_MAIL.  Any of the possible
-## facilities listed in the syslog(3C) manpage may be used here except
+## facilities listed in the syslog(3) manpage may be used here except
 ## LOG_KERN.
 #
 syslog_facility LOG_MAIL
@@ -130,9 +130,8 @@
 ## This configuration option is provided as a way to limit the verbosity
 ## of imapproxy.  If no value is specified, it will default to no priority
 ## mask and you'll see all possible log messages.  Any of the possible
-## priority values listed in the syslog(3C) manpage may be used here.  By
-## default, I've left this commented out so you will see all possible log
-## messages.
+## priority values listed in the syslog(3) manpage may be used here.  By
+## default, you will see all possible log messages.
 #
 #syslog_prioritymask LOG_WARNING
 
@@ -221,8 +220,8 @@
 
 
 #
-## Set any of these to "yes" if the corresponding TLS
-## version is not sufficiently secure for your needs
+## Set any of these to "yes" if the corresponding TLS version is not
+## sufficiently secure for your needs.
 #
 #tls_no_tlsv1 no
 #tls_no_tlsv1.1 no
@@ -265,5 +264,3 @@
 ## Default if unset is AF_UNSPEC for both A and AAAA.
 #
 #ipversion_only 6
- 
-
