Index: up-imapproxy-1.2.7rc2/include/imapproxy.h
===================================================================
--- up-imapproxy-1.2.7rc2.orig/include/imapproxy.h	2009-10-16 12:49:51.000000000 +0200
+++ up-imapproxy-1.2.7rc2/include/imapproxy.h	2009-10-16 12:52:34.000000000 +0200
@@ -196,6 +196,10 @@
 #define LOGIN_DISABLED          1
 #define LOGIN_NOT_DISABLED      0
 
+
+#define DEFAULT_SERVER_CONNECT_RETRIES	10
+#define DEFAULT_SERVER_CONNECT_DELAY	5
+
 /*
  * One IMAPServerDescriptor will be globally allocated such that each thread
  * can save the time of doing host lookups, service lookups, and filling
@@ -361,6 +365,7 @@
 extern void ICC_Recycle_Loop( void );
 extern void LockMutex( pthread_mutex_t * );
 extern void UnLockMutex( pthread_mutex_t * );
+extern void SetDefaultConfigValues(ProxyConfig_Struct *);
 extern void SetConfigOptions( char * );
 extern void SetLogOptions( void );
 extern int Handle_Select_Command( ITD_Struct *, ITD_Struct *, ISC_Struct *, char *, int );
Index: up-imapproxy-1.2.7rc2/src/config.c
===================================================================
--- up-imapproxy-1.2.7rc2.orig/src/config.c	2009-10-16 12:49:51.000000000 +0200
+++ up-imapproxy-1.2.7rc2/src/config.c	2009-10-16 12:51:35.000000000 +0200
@@ -125,7 +125,6 @@
 static void SetNumericValue( char *, int *, unsigned int );
 static void SetBooleanValue( char *, unsigned int *, unsigned int );
 
-
 /*
  * An array of Config_Structs will need to be allocated, one for
  * every possible keyword/value combination (plus a NULL).  Note that
@@ -320,7 +319,25 @@
 }
 
 
+/*++
+ * Function:	SetDefaultConfigValues
+ *
+ * Purpose:	Set global configuration default values
+ *
+ * Parameters:	pointer to global config struct
+ *
+ * Returns:	nada.
+ *
+ * Authors:	Jose Luis Tallon <jltallon@adv-solutions.net>
+ *--
+ */
+void SetDefaultConfigValues(ProxyConfig_Struct *PC_Struct)
+{
+    PC_Struct->server_connect_retries = DEFAULT_SERVER_CONNECT_RETRIES;
+    PC_Struct->server_connect_delay = DEFAULT_SERVER_CONNECT_DELAY;
 
+    return;
+}
 
 /*++
  * Function:	SetConfigOptions
Index: up-imapproxy-1.2.7rc2/src/main.c
===================================================================
--- up-imapproxy-1.2.7rc2.orig/src/main.c	2009-10-16 12:49:51.000000000 +0200
+++ up-imapproxy-1.2.7rc2/src/main.c	2009-10-16 12:49:51.000000000 +0200
@@ -363,7 +363,8 @@
 	syslog( LOG_INFO, "%s: Using default configuration file '%s'.",
 		fn, ConfigFile );
     }
-    
+
+    SetDefaultConfigValues(&PC_Struct);
     SetConfigOptions( ConfigFile );
     SetLogOptions();
 
